most_executed_script.gs

```diff
@@ -0,0 +1,166 @@
+          resultData.entries[i].labels['script.googleapis.com/process_id'];
+        var projectId =
+          resultData.entries[i].labels['script.googleapis.com/project_key'];
+        processIdsMap[processId] = projectId;
+      }
+    }
+    if (limit) {
+      break;
+    }
+    var header = {
+      'Authorization': 'Bearer ' + ScriptApp.getOAuthToken()
+    };
+    var body = {
+      'projectIds': [
+        cloudProjectId
+      ],
+      'resourceNames': [
+        'projects/' + cloudProjectId
+      ],
+      'pageToken': pageToken,
+      'orderBy': 'timestamp desc'
+    };
+    var options = {
+      'method': 'post',
+      'contentType': 'application/json',
+      'headers': header,
+      'payload': JSON.stringify(body),
+      'muteHttpExceptions': false
+    };
+    var url = 'https://logging.googleapis.com/v2/entries:list';
+    var response = UrlFetchApp.fetch(url, options);
+    var json = response.getContentText();
+    resultData = JSON.parse(json);
+    pageToken = resultData.nextPageToken;
+  }
+
+  // go through all the processIds and count the different projectIds
+  for (i in processIdsMap) {
+    if (projectIdsMap[processIdsMap[i]]) {
+      projectIdsMap[processIdsMap[i]] += 1;
+    } else {
+      projectIdsMap[processIdsMap[i]] = 1;
+    }
+  }
+  return projectIdsMap;
+}
+
+/**
+* Get the maximum executed Script from all the cloud projects
+* @param {Date} fromTime is the date and time from 
+* which executions has to be seen
+* @param {String} projectType is the enum having values 
+* {SYSTEM_PROJECT, ALL_PROJECT, SPECIFIC_PROJECT, CUSTOM_PROJECT}
+* @param {string} cloudProjectId is cloud project id of specific project 
+* otherwise null
+* @return {Object} Array of projectIds along with the number of executions 
+* of the executed scripts and the cloud project id
+*/
+function getMostExecutedScriptFromAllCloudProjects(
+  fromTime, projectType, cloudProjectId) {
+  var processIdWithExecutions = {};
+  var processIdWithGCPId = {};
+  if (projectType == "SPECIFIC_PROJECT") {
+    var projDetails = getProjectDetails(cloudProjectId);
+    if (projDetails != null) {
+      var apiEnabled = enableLogginApisPvt(projDetails.projectNumber);
+      if (apiEnabled) {
+        var result = getNumberOfExecutionOfScript(cloudProjectId, fromTime);
+        for (var j in result) {
+          if (processIdWithExecutions[j]) {
+            processIdWithExecutions[j] += result[j];
+          } else {
+            processIdWithExecutions[j] = result[j];
+          }
+          processIdWithGCPId[j] = cloudProjectId;
+        }
+      }
+    }
+  } else {
+    var allProjects = listAllCloudProjects();
+    var i;
+    var max = 0;
+
+    // loop for all thr projects and get most executed scripts of each project
+    for (i = 0; i < allProjects.length; i++) {
+      if (allProjects[i].lifecycleState != 'ACTIVE') {
+        continue;
+      }
+      if (JSON.stringify(allProjects[i].projectId, null, 2).indexOf("sys") ==
```

Comment:
add a comment explaining why you did this.

Reply:
Updated